src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryColor.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Core.IR.Nodes;
+using Microsoft.PowerFx.Core.Types.Enums;
+using Microsoft.PowerFx.Interpreter;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Functions
+{
+    internal partial class Library
+    {
+        // ColorTable is ARGB
+        private static readonly Regex RegexColorTable = new (@"^#(?<a>[0-9a-fA-F]{2})(?<r>[0-9a-fA-F]{2})(?<g>[0-9a-fA-F]{2})(?<b>[0-9a-fA-F]{2})?$", RegexOptions.Compiled);
+
+        // CSS format is RGBA
+        private static readonly Regex RegexCSS = new (@"^#(?<r>[0-9a-fA-F]{2})(?<g>[0-9a-fA-F]{2})(?<b>[0-9a-fA-F]{2})(?<a>[0-9a-fA-F]{2})?$", RegexOptions.Compiled);
+        
+        public static FormulaValue ColorValue(IRContext irContext, StringValue[] args)
+        {
+            var val = args[0].Value;
+
+            if (string.IsNullOrEmpty(val))
+            {
+                return new BlankValue(irContext);
+            }
+
+            Match match;
+            if (ColorTable.InvariantNameToHexMap.ContainsKey(val))
+            {
+                var hexStringColor = string.Format("#{0:X8}", ColorTable.InvariantNameToHexMap[val]);
+                match = RegexColorTable.Match(hexStringColor);
+            }
+            else
+            {
+                match = RegexCSS.Match(val);
+            }
+
+            if (match.Success)
+            {
+                var r = byte.Parse(match.Groups["r"].Value, System.Globalization.NumberStyles.HexNumber);
```

Comment:
> byte.Parse(match.Groups["r"].Value, System.Globalization.NumberStyles.HexNumber); [](http://example.com/codeflow?start=23&length=82)

nit - this is repeated 4x, could be a static helper

Reply:
updated.